lib/gst/player/gstplayer.c

```diff
@@ -1121,6 +1144,458 @@ element_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
   }
 }
 
+static void
+player_set_flag (GstPlayer *self, gint pos)
+{
+  gint flags;
+
+  g_object_get (self->priv->playbin, "flags", &flags, NULL);
+  flags |= pos;
+  g_object_set (self->priv->playbin, "flags", &flags, NULL);
+}
+
+static void
+player_clear_flag (GstPlayer *self, gint pos)
+{
+  gint flags;
+
+  g_object_get (self->priv->playbin, "flags", &flags, NULL);
+  flags &= ~pos;
+  g_object_set (self->priv->playbin, "flags", &flags, NULL);
+}
+
+static gboolean
+is_track_enabled (GstPlayer *self, gint pos)
+{
+  gint flags;
+
+  g_object_get (self->priv->playbin, "flags", &flags, NULL);
+  if (flags & pos)
```

Comment:
I think this gives a compiler warning about missing parenthesis. Make this if ((flags & pos))